src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_pending_resumption_handler.cc

```diff
@@ -0,0 +1,253 @@
+  } else {
+    LOG4CXX_DEBUG(logger_,
+                  "Resumption of subscriptions is NOT successful"
+                      << " module type: " << module_uid.first
+                      << " module id: " << module_uid.second);
+    ProcessNextFreezedResumption(module_uid);
+  }
+}
+
+void RCPendingResumptionHandler::HandleResumptionSubscriptionRequest(
+    application_manager::AppExtension& extension,
+    resumption::Subscriber& subscriber,
+    application_manager::Application& app) {
+  UNUSED(extension);
+  LOG4CXX_AUTO_TRACE(logger_);
+  sync_primitives::AutoLock lock(pending_resumption_lock_);
+  const uint32_t app_id = app.app_id();
+  LOG4CXX_TRACE(logger_, "app id " << app_id);
+
+  auto rc_extension = RCHelpers::GetRCExtension(app);
+  auto subscriptions = rc_extension->InteriorVehicleDataSubscriptions();
+
+  std::vector<ModuleUid> already_pending;
+  std::vector<ModuleUid> need_to_subscribe;
+  for (auto subscription : subscriptions) {
+    if (IsPending(subscription)) {
+      already_pending.push_back(subscription);
+    } else {
+      need_to_subscribe.push_back(subscription);
+    }
+  }
+
+  for (auto subscription : already_pending) {
+    const auto cid = application_manager_.GetNextHMICorrelationID();
+    const auto subscription_request =
+        CreateSubscriptionRequest(subscription, cid);
+    const auto fid = GetFunctionId(subscription_request);
+    const auto resumption_request =
+        MakeResumptionRequest(cid, fid, *subscription_request);
+    freezed_resumptions_[subscription].push(resumption_request);
+    subscriber(cid, resumption_request);
+    LOG4CXX_DEBUG(logger_,
+                  "Freezed request with correlation_id: "
+                      << cid << " module type: " << subscription.first
+                      << " module id: " << subscription.second);
+  }
+
+  for (auto subscription : need_to_subscribe) {
+    const auto cid = application_manager_.GetNextHMICorrelationID();
+    const auto subscription_request =
+        CreateSubscriptionRequest(subscription, cid);
+    const auto fid = GetFunctionId(subscription_request);
+    const auto resumption_request =
+        MakeResumptionRequest(cid, fid, *subscription_request);
+    pending_requests_.emplace(cid, *subscription_request);
+    subscriptions_.emplace_back(subscription);
+    subscribe_on_event(fid, cid);
+    subscriber(cid, resumption_request);
+    LOG4CXX_DEBUG(logger_,
+                  "Sending request with correlation id: "
+                      << cid << " module type: " << subscription.first
+                      << " module id: " << subscription.second);
+    application_manager_.GetRPCService().ManageHMICommand(subscription_request);
+  }
+}
+
+void RCPendingResumptionHandler::ClearPendingResumptionRequests() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  sync_primitives::AutoLock lock(pending_resumption_lock_);
+
+  while (!pending_requests_.empty()) {
+    auto data_request = pending_requests_.front();
```

Comment:
If some resumption is reverting is not a reason to remove all pending requests, pending requests may contain data for other resumptions.

Reply:
@LuxoftAKutsan 
Fixed in:https://github.com/LuxoftSDL/sdl_core/pull/99/commits/114a2b4f4858dd189f26c13f00aa4292489b374b